Admin dashboard vendor list react

Aunshon opened the pull request:
### All Submissions:

* [x] My code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)
* [x] My code satisfies feature requirements
* [x] My code is tested
* [x] My code passes the PHPCS tests
* [x] My code has proper inline documentation
* [x] I've included related pull request(s) (optional)
* [x] I've included developer documentation (optional)
* [x] I've added proper labels to this pull request

### Related Pull Request(s)

* https://github.com/getdokan/dokan-pro/pull/4884


### Closes 

* Closes https://github.com/getdokan/dokan-pro/issues/4668


### How to test the changes in this Pull Request:

* As mentioned in issue.


### Changelog entry
```text
- **update:** Admin Dashboard vendor list UI with Modern Design and Improved User Experience.
```

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Vendors admin page: list, search, status tabs, filters, bulk approve/disable, per-row actions, pagination and status counts.
  - Withdraws admin page & reporting: rich table with actions, modals, bulk workflows and PayPal export with polling; new REST export endpoints for reports.

- **UI / Components**
  - New reusable AdminDataView table, tabs, filter panel, empty-state components and updated admin styles.

- **Improvements**
  - Currency formatting now respects admin/dashboard currency settings.

- **Documentation**
  - Guides for custom export controllers and AdminDataViewTable usage.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Aunshon committed:
feat: add DokanSelect component with customizable styles and indicators

Aunshon committed:
enhance: update DokanSelect component to improve cursor styles for indicators and options

Aunshon committed:
enhance: update indicator separator visibility in DokanSelect component

Aunshon committed:
enhance: improve component composition and style precedence in DokanSelect

Aunshon committed:
feat: add DokanAsyncSelect and VendorAsyncSelect components for enhanced vendor selection

Aunshon committed:
feat: add ProductAsyncSelect component for enhanced product selection

Aunshon committed:
feat: add WcDateRangePicker component for date range selection

Aunshon committed:
feat: add WpDateTimePicker component for date and time selection

Aunshon committed:
feat: refactor ProductAsyncSelect for improved loading and option handling

Aunshon committed:
feat: add OrderAsyncSelect component for enhanced order selection

Aunshon committed:
feat: add SearchInput component for enhanced search functionality

Aunshon committed:
feat: add CouponAsyncSelect component for coupon selection

Aunshon committed:
feat: add Test component with various Dokan UI elements and functionality

Aunshon committed:
feat: remove DokanSelect export from index.tsx

Aunshon committed:
feat: remove DokanSelect component from Test.tsx

Aunshon committed:
feat: enhance Test component with ProductAsyncSelect and VendorAsyncSelect

Aunshon committed:
feat: remove unused imports from Test component

Aunshon committed:
feat: rename DokanAsyncSelect to AsyncSelect and update imports

Aunshon committed:
feat: rename DokanAsyncSelect to AsyncSelect and update imports

Aunshon committed:
feat: add Select component and update AsyncSelect usage in Test component

Aunshon committed:
feat: refactor AsyncSelect and related components to use BaseSelectProps for improved type safety

Aunshon committed:
feat: enable multi-select functionality in Test component

Aunshon committed:
feat: integrate SimpleInput for date input in WcDateRangePicker, WpDatePicker, and WpDateTimePicker components

Aunshon committed:
feat: add documentation for AsyncSelect, CouponAsyncSelect, OrderAsyncSelect, ProductAsyncSelect, VendorAsyncSelect, WpDateTimePicker, SearchInput, and Select components

Aunshon committed:
Merge branch 'develop' into feat/dokan-select-box

Aunshon committed:
feat: add prefetch and strict validation options to AsyncSelect components

Aunshon committed:
feat: update AsyncSelect components to improve fetching logic and add unique instance IDs

Aunshon committed:
feat: enhance AsyncSelect components with unique merging logic for prefetched options

Aunshon committed:
feat: implement skip merging logic for AsyncSelect components on dependency changes

Aunshon committed:
feat: replace Test component with Status component in routing

Aunshon committed:
feat: add Vendors management page and related components

Aunshon committed:
Merge branch 'feat/dokan-select-box' into update/dokan-vendor-admin-list

Aunshon committed:
feat: enhance vendor admin list with avatar display and action buttons

Aunshon committed:
feat: update vendor admin list with improved URL handling and status display

Aunshon committed:
feat: add confirmation modal for vendor approval and disable actions

Aunshon committed:
feat: add selection state management for bulk actions in vendor admin list

Aunshon committed:
feat: rename WcDateRangePicker and WpDateTimePicker components to DateRangePicker and DateTimePicker

Aunshon committed:
feat: enhance AsyncSelect components with Tailwind CSS styling and improve option label formatting

Aunshon committed:
feat: update documentation for Props API and enhance WpDatePicker with unique instance IDs

Aunshon committed:
feat: enhance AsyncSelect component with icon support and update documentation

Aunshon committed:
feat: update ProductAsyncSelect endpoint and enhance SearchInput with stable debouncing

Aunshon committed:
feat: remove onClose handler from DateRangePicker component

Aunshon committed:
feat: replace Test component with Status in Dokan routes

Aunshon committed:
Finish select components

Aunshon committed:
feat: enhance AsyncSelect components with onMenuOpen handling and loading optimization

Aunshon committed:
Merge branch 'develop' into feat/dokan-select-box

Aunshon committed:
Merge branch 'feat/dokan-select-box' into update/dokan-vendor-admin-list

Aunshon committed:
feat: add filter button and toggle functionality to Vendors page

Aunshon committed:
feat: add seller badge and subscription filters to Vendors page

Aunshon committed:
feat: improve loading states and optimize rendering for vendor fields

Aunshon committed:
feat: add SellerBadgeSelect component and update vendor list layout

Aunshon committed:
feat: enhance vendor filtering with dynamic fields and improved query handling

Aunshon committed:
Merge branch 'develop' into update/dokan-vendor-admin-list

Aunshon committed:
Merge branch 'develop' into update/dokan-vendor-admin-list

# Conflicts:
#	includes/DependencyManagement/Providers/AdminDashboardServiceProvider.php
#	src/admin/dashboard/components/Dashboard.tsx

Aunshon committed:
Merge branch 'feat/dokan-widthraw-admin-list' into update/dokan-vendor-admin-list

# Conflicts:
#	includes/DependencyManagement/Providers/AdminDashboardServiceProvider.php
#	src/admin/dashboard/components/Dashboard.tsx

Aunshon committed:
Merge branch 'feat/dokan-widthraw-admin-list' into update/dokan-vendor-admin-list

Aunshon committed:
refactor: enhance vendor admin list with dynamic filters and improved UI elements

- Updated filters to allow dynamic field injection and better customization.
- Improved UI for status tabs and filters with consistent theming.
- Refactored filter logic and replaced redundant methods for better clarity.
- Integrated search and tab selection into the main table component for enhanced usability.

Aunshon committed:
refactor: remove unused components and redundant state logic in Vendors page

- Removed `DokanTab` and `Filter` components.
- Deleted `showFilters` state and unnecessary `useEffect` for filter visibility.
- Simplified and cleaned up markup and filter methods for improved readability.

Aunshon committed:
Merge branch 'develop' into update/dokan-vendor-admin-list

Aunshon committed:
Merge branch 'feat/dokan-widthraw-admin-list' into update/dokan-vendor-admin-list

Aunshon committed:
feat: add 'vendors' to supported keys in PanelSwitch filter

MdAsifHossainNadim committed:
Merge branch 'feat/dokan-widthraw-admin-list' into update/dokan-vendor-admin-list

# Conflicts:
#	src/admin/panel-switcher/PanelSwitch.tsx

MdAsifHossainNadim committed:
refactor: Remove unused `shouldRender` state and legacy redirect logic from PanelSwitch and LegacySwitcher

Aunshon committed:
fix: update vendor navigation path in Vendors page

mrabbani commented on includes/Admin/Dashboard/Dashboard.php:
```diff
@@ -336,7 +337,20 @@ protected function register_admin_dashboard_scripts() {
+                    'upgradeToPro'      => 'https://dokan.co/wordpress/upgrade-to-pro/?utm_source=plugin&utm_medium=wp-admin&utm_campaign=dokan-lite',
+                    'dummy_data'        => DOKAN_PLUGIN_ASSEST . '/dummy-data/dokan_dummy_data.csv',
+                    'adminOrderListUrl' => OrderUtil::get_admin_order_list_url(),
+                    'adminOrderEditUrl' => OrderUtil::get_admin_order_edit_url(),
```
replace the `wp_localize_script` with the `wp_add_inline_scripts`

mrabbani commented on src/admin/dashboard/pages/vendors.tsx:
```diff
@@ -0,0 +1,793 @@
+        try {
+            // Use a stable filter id to let pro register fields
+            // @ts-ignore
+            const injected = wp.hooks.applyFilters(
```
Pls use from the package.

Aunshon committed:
refactor: replace `wp.hooks.applyFilters` with `applyFilters` and update `wp_localize_script` usage

- Used `@wordpress/hooks`'s `applyFilters` for consistency.
- Replaced `wp_localize_script` with `wp_add_inline_script` for better script handling.
